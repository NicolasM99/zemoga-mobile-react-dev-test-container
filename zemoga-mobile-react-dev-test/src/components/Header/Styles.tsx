import { StyleSheet } from 'react-native';

export const fontSizes = {
  h1: 20,
  h2: 18,
  h3: 16,
  h4: 14,
  h5: 12,
  h6: 10
};

export const headerStyles = StyleSheet.create({
  font: {
    marginBottom: 16
  }
});
