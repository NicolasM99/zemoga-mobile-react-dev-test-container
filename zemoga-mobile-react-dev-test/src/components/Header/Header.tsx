import React, { FC } from 'react';

import { Text } from 'react-native';

import { IHeader } from './@types/header';
import { fontSizes, headerStyles } from './Styles';

const Header: FC<IHeader> = ({ children, bold = true, variant = 'h1', style = {} }: IHeader) => {
  return (
    <Text
      style={{
        fontSize: fontSizes[variant],
        fontWeight: bold ? 'bold' : 'normal',
        ...headerStyles.font,
        ...style
      }}
    >
      {children}
    </Text>
  );
};

export default Header;
