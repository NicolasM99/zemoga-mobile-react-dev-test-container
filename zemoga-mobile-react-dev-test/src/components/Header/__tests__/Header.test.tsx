import { render, screen } from '@testing-library/react-native';

import Header from '../Header';

describe('<Header /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<Header>Header test</Header>);
    expect(screen).toMatchSnapshot();
  });

  it('Should render a header with correct label', async () => {
    render(<Header>Header test</Header>);
    const header = await screen.findByTestId('header');
    expect(header.props.children).toEqual('Header test');
  });
});
