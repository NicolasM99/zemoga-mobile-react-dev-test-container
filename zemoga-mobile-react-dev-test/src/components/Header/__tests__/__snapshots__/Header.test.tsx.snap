// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> unit tests Should mount correctly the component 1`] = `
<Text
  style={
    Object {
      "fontSize": 20,
      "fontWeight": "bold",
      "marginBottom": 16,
    }
  }
  testID="header"
>
  Header test
</Text>
`;
